### PR TITLE
Micro-climate

### PR DESCRIPTION
**Pull request by xuanluo113**
Enhancement for Simulation using Urban Microclimate Conditions
---------------------
With the global trend towards urbanization, energy use and the associated GHG emissions are growing significantly in cities. Reducing energy use in the building sector is a key strategy for cities to achieve their energy and environmental goals. The urban climate and microclimate can strongly affect the building energy demand. As air temperatures in urban areas are higher due to the urban heat island (UHI) effect and the wind speeds can be lower due to wind sheltering, the microclimate in urban areas differs significantly from the climate in rural areas. The proposed enhancement intends to enable EnergyPlus simulations to use the local ambient conditions, which can be from measurements or CFD simulation, including parameters of outdoor air velocity and direction, dry-bulb temperature and humidity.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Code style (parentheses padding, variable names)
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check -- verify this
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add rules to spreadsheet
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Documentation changes in place
 - [ ] Changed docs build successfully
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

File: design/FY2017/NFP-Enhancement for Urban Microclimate.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+![EplusUrban](EnergyPlusUrbanClimate.png)
+
+***Figure 1 EnergyPlus data exchange with urban microclimate model***
+
+The current implementation of EnergyPlus assumes a default vertical temperature gradient. It does not provide a direct link to allow data exchange with urban microclimate models to override local environmental data at the zone and surface levels, such as the outdoor air temperature, humidity, wind velocity and direction. However, there are reportable surface and zone outdoor conditions tracked separately as:
```
Comment (macumber) [inline design/FY2017/NFP-Enhancement for Urban Microclimate.md:17]: Why is this problem being addressed with EMS sensors only (which can be actuated to include input from external preprocessed CFD simulations) but the external shading and LWR proposals have input objects which reference large amounts of data (e.g. CSV files with shading coefficients and surface temperatures).  This seems like it sets up two different paradigms (large amounts of input data vs run time coupling via EMS actuators) for coupling E+ with other calculation engines (shading or CFD).  What is the rationale for the difference in approach?  Is the idea that CFD is more coupled to the building model and thus requires co-simulation?  What about LWR, isn't that coupled as well?

File: design/FY2017/NFP-Enhancement for Urban Microclimate.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+# References
+
+n/a
+
+# E-mail comments
+
+**Lixing Gu <gu@fsec.ucf.edu> Thu, Jun 1, 2017** 
+
+LocalOutdoorAir:Node
+Since you require to input outdoor conditions, T, W, Wind Speed and direction, is it possible to add solar radiation treated as diffuse component?
```
Comment (macumber) [inline design/FY2017/NFP-Enhancement for Urban Microclimate.md:67]: Couldn't Gu's suggestion be extended to cover the external shading calculation feature proposal without requiring additional input objects?

Comment (macumber) [reply design/FY2017/NFP-Enhancement for Urban Microclimate.md:67]: Could it also be extended to cover the LWR feature proposal?

File: design/FY2017/NFP-Enhancement for Urban Microclimate.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+Rather than add the concept of outdoor air nodes with one-to-one correspondence with surfaces and zones, here it may be better to just add new EMS actuators.  Surface and zone outdoor conditions are already tracked separately, and they are reportable:
+	Zone,Average,Surface Outside Face Outdoor Air Drybulb Temperature [C]
+	Zone,Average,Surface Outside Face Outdoor Air Wetbulb Temperature [C]
+	Zone,Average,Surface Outside Face Outdoor Air Wind Speed [m/s]
+	Zone,Average,Zone Outdoor Air Drybulb Temperature [C]
+	Zone,Average,Zone Outdoor Air Wetbulb Temperature [C]
+	Zone,Average,Zone Outdoor Air Wind Speed [m/s]
+
+So it would be straightforward to add new EMS actuators for these six variables to override these values directly.  The only thing missing is wind direction, which could be added to the same data structures.
```
Comment (macumber) [inline design/FY2017/NFP-Enhancement for Urban Microclimate.md:81]: How will E+ use wind direction?  Don't you really want to know pressure at the surface so you can calculate infiltration?  Wouldn't pressure be available from a CFD calculation?

**Comment by nrel-bot-3** [comment]
@hongtz68 @XuanLuoLBL @lgentile it has been 17 days since this pull request was last updated.

**Pull request closed without merging**